strawberryfields/utils/post_processing.py

```diff
@@ -0,0 +1,341 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module provides functions for post-processing samples.
+"""
+import numpy as np
+
+
```

Comment:
It might be good to move the smaller utility functions first/to the top of the file. This way, it's easier to read the file; by the time you get to the functions that use them, you know what they're doing :)

For example, maybe the following order makes sense?

1. First, general validation functions:

   * `_check_samples`
   * `_check_modes`

2. Then, auxiliary functions used in other post-processing functions:

   * `_product_for_modes`
   * `_sample_variance`
   * `_sample_expectation`

3. Finally, user-facing functions:

   *  `number_expectation_pnr`
   * `number_variance_pnr`
   * `quadrature_variance_homodyne`
   * `all_fock_probs_pnr`

Reply:
sure, reorganized them!